### ISSUE TITLE
Insiders API Change: no NotebookData class in NotebookSerializer.serializeNotebook used

### ISSUE THREAD

**Issue by AnWeber** [microsoft/vscode#232342]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.95.0-insider (user setup)
Commit: 38dc6ac5a771cc94bde1344722bb2d02c80096ea
Date: 2024-10-25T22:08:52.005Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.22631

Steps to Reproduce:

1. implement NotebookSerializer interface in an extension
2. check first args of serializeNotebook with `notebookData instanceOf vscode.NotebookData`
3. with newest VSCode Insiders this checks returns false. with latestet VSCode this checks returns true

There were changes in NoteBookSerializer, maybe these produce this changed behaviour: https://github.com/microsoft/vscode/commit/c4645ea0ec0093893e7c16b1f628128d439da671#diff-b9e56d2e3070ee2be4079f435bcc2c5ea8c9c96c0ab9744128d25e337952a18e

Workaround: I willl implement a typeguard to check if object matches API of NotebookData Class

Code which produces this error: https://github.com/AnWeber/httpbook/blob/main/src/extension/notebook/httpNotebookSerializer.ts#L190
Bug of User with Insiders: https://github.com/AnWeber/httpbook/issues/134

**Comment by DonJayamanne** [comment]
@rebornix Would this have something to do with Remote Save.
@AnWeber Thank you for filing this issue, however the changes you highlighted are not related, as those changes are in a built in extension and not the API for other extensions (i.e. ipynb is an extension that is shipped with VS Code for serializing of Jupyter *.ipynb notebook files)

I'll have a look and see whats going on here.

**Comment by DonJayamanne** [comment]
> . with latestet VSCode this checks returns true


@AnWeber Here I'm assuming you are referring to 1.94 of VS Code

**Comment by DonJayamanne** [comment]
@rebornix Confirmed, this is due to the change related to remote save.

**Comment by AnWeber** [comment]
@DonJayamanne yes, I meant 1.94. I think the ticket can also be closed. The code is already productive. In other words, the relevant areas should already be fixed in the extensions. I just wanted to point out the changed behavior in the Insiders
